e2e_tests/tests/experiment/test_custom_searcher.py

```diff
@@ -582,3 +584,102 @@ def save_state(self, experiment_id: int, operations: List[searcher.Operation]) -
+
+    projects = []
+    experiments = []
+    for i in workspaces:
+        body1 = bindings.v1PostProjectRequest(name=f"p_{uuid.uuid4().hex[:8]}", workspaceId=i.id)
+        pid1 = bindings.post_PostProject(
+            session,
+            body=body1,
+            workspaceId=i.id,
+        ).project.id
+
+        body2 = bindings.v1PostProjectRequest(name=f"p_{uuid.uuid4().hex[:8]}", workspaceId=i.id)
+        pid2 = bindings.post_PostProject(
+            session,
+            body=body2,
+            workspaceId=i.id,
+        ).project.id
+
+        projects.append(pid1)
+        projects.append(pid2)
+
+        experiments.append(
+            exp.create_experiment(
+                conf.fixtures_path("no_op/single.yaml"),
+                conf.fixtures_path("no_op"),
+                ["--project_id", str(pid1)],
+            )
+        )
+        experiments.append(
+            exp.create_experiment(
+                conf.fixtures_path("no_op/single.yaml"),
+                conf.fixtures_path("no_op"),
+                ["--project_id", str(pid1)],
+            )
+        )
+        experiments.append(
+            exp.create_experiment(
+                conf.fixtures_path("no_op/single.yaml"),
+                conf.fixtures_path("no_op"),
+                ["--project_id", str(pid2)],
+            )
+        )
+        experiments.append(
+            exp.create_experiment(
+                conf.fixtures_path("no_op/single.yaml"),
+                conf.fixtures_path("no_op"),
+                ["--project_id", str(pid2)],
+            )
+        )
+
+    for x in experiments:
+        exp.wait_for_experiment_state(experiment_id=x, target_state=experimentv1State.COMPLETED)
+    bindings.post_KillExperiments(
+        session, body=bindings.v1KillExperimentsRequest(experimentIds=experiments)
+    )
+
+    bindings.post_ArchiveExperiment(session, id=experiments[0])
+    bindings.post_ArchiveExperiment(session, id=experiments[3])
+    bindings.post_ArchiveExperiment(session, id=experiments[5])
+    bindings.post_ArchiveExperiment(session, id=experiments[6])
+
+    archived_exp = [experiments[0], experiments[3], experiments[5], experiments[6]]
+
+    r1 = bindings.get_GetExperiments(session, archived=False)
+
+    for e in r1.experiments:
+        assert e.id not in archived_exp
+
+    bindings.post_ArchiveProject(session, id=projects[1])
+    bindings.post_ArchiveProject(session, id=projects[2])
+
+    archived_exp.append(experiments[2])
+    archived_exp.append(experiments[4])
+
+    r2 = bindings.get_GetExperiments(session, archived=False)
+    for e in r2.experiments:
+        assert e.id not in archived_exp
+
+    bindings.post_ArchiveWorkspace(session, id=workspaces[1].id)
+
+    archived_exp.append(experiments[7])
+
+    r3 = bindings.get_GetExperiments(session, archived=False)
+    for e in r3.experiments:
+        assert e.id not in archived_exp
+
+    for w in workspaces:
+        bindings.delete_DeleteWorkspace(session, id=w.id)
```

Comment:
could you please add a test that checks fetching experiments within an archived project using the projectId parameter